packages/insights/README.md

```diff
@@ -0,0 +1,201 @@
+make TypeScript happy). If implemented correctly, it should log all navigation route changes to
+Application Insights!
+
+```tsx
+export default function Navigation() {
+    const navigationRef = useNavigationContainerRef();
+    const routeNameRef = useRef<string | undefined>();
+
+    return (
+        <NavigationContainer
+            ref={navigationRef}
+            onReady={() => {
+                const currentRoute = navigationRef.getCurrentRoute();
+                if (!currentRoute) return;
+                routeNameRef.current = currentRoute.name;
+            }}
+            onStateChange={async () => {
+                const currentRoute = navigationRef.getCurrentRoute();
+                if (!currentRoute) return;
+                const previousRouteName = routeNameRef.current;
+                const currentRouteName = currentRoute.name;
+
+                if (previousRouteName !== currentRouteName) {
+                    trackNavigation(currentRouteName);
+                }
+
+                routeNameRef.current = currentRouteName;
+            }}
+        >
+            <RootNavigator />
+        </NavigationContainer>
+    );
+}
+```
+
+**Custom tracking**
+
+If you need to add additional tracking to your app, we have some helper methods for that! The
+easiest way to track something is to use `trackCustom`, which takes a string for the title, and
+optionally an object with data. `trackCustom` logs to both long term log and short term log. If you
+want more control over how the tracking should occur, use `track` if you want both long term and
+short term tracking, or `trackShortTerm` and `trackLongTerm` respectively.
+
+Example:
+
+```tsx
+//this is the easiest way to add some custom tracking
+trackCustom("Custom event!", { param1: "param1", param2: "param2" });
+
+//this is identical to the line above
+track(metricKeys.CUSTOM, undefined, "Custom event!", { param1: "param1", param2: "param2" });
+
+//this will only log to the short term log
+trackShortTerm(metricKeys.CUSTOM, undefined, "Custom event!", {
+    param1: "param1",
+    param2: "param2",
+});
+
+//this will only log to the long term log
+trackLongTerm(metricKeys.CUSTOM, undefined, "Custom event!", {
+    param1: "param1",
+    param2: "param2",
+});
+```
+
+### Adding common information to all events
+
+If you want to add common telemetry to all events, use the `addTelemetryInitializer` function.
+
+Usage of `addTelemetryInitializer` can be found
+[here](https://github.com/microsoft/ApplicationInsights-JS#telemetry-initializers). You can add
+multiple envelopes to our application insights instances.
+
+Import the function from `@equinor/mad-insights` in order to use our own instances of application
+insights.
+
+```tsx
+import { ITelemetryItem, addTelemetryInitializer, appInsightsInit } from "@equinor/mad-insights";
+import * as APP from "./app.json";
+
+
+export default function App() {
+    useEffect(() => {
```

Comment:
For initializing app insights, that might be a good idea!

For telemetry stuff I don't think that fits, Since you might add telemetry initializers in other parts of the app (e.g. Department ID telemetry must be added after login)

Reply:
`useAppInsights` added as a hook, will initialize app insights for you